### ISSUE TITLE
Enable required Axe rules for Section 508 accessibility compliance

### ISSUE THREAD

**Issue by alisan16** [department-of-veterans-affairs/caseflow#12884]
#12742 updated the Sniffybara and Axe versions that run accessibility scans in the codebase, which are run on rspec feature tests. In order to meet Section 508 compliance, the Caseflow rspec tests should run cleanly with all `section 508`, `wcag2aa` and `wcag2a` rules enabled. [Axe rule descriptions can be found here](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md)

## Background/context/resources
Approximate incidence rates for each failed rule can be found in the table below. For the rules that have the most number of errors, it is likely a common UI component that is the issue. Check the caseflow-frontend-toolkit repo.

| Axe rule-id | # of Errors |
|------------|------------|
| aria-roles | 760 |
| definition-list | 47 |
| aria-required-parent | 22 |
| listitem | 17 |
| aria-allowed-attr | 14 |
| duplicate-id-aria | 8 |
| aria-input-field-name | 6  |
| aria-required-children | 6 |
| list | 4 |
| color-contrast | 2  |
| scrollable-region-focusable | 1 |
| duplicate-id-active | |

## Technical notes
1. To get started, enable the rule by [setting `"enabled": true` on the relevant rule](https://github.com/department-of-veterans-affairs/caseflow/pull/14063/commits/60ea1d384a63beaa9e255c73f581c50a6b93eac3) or removing the relevant rule from the `rules` object in spec/support/VA-axe-run-configuration.json (rules are enabled by default) and submit a PR to kickstart the CircleCI process of running the tests.
  a. You can also run the rule locally by running `bundle exec rspec` to get a sense of where the failures are. This takes a long time to run, so feel free to cancel once a few errors pop up.
2. Investigate the components which are throwing the Sniffybara errors and fix them for compliance.
3. Once the `rspec` tests complete without failure, you've succeeded!

**Comment by hschallhorn** [comment]
Looks like echo grabbed the first of these. Highly recommend starting by [enabling the rule](https://github.com/department-of-veterans-affairs/caseflow/pull/14063/commits/60ea1d384a63beaa9e255c73f581c50a6b93eac3) and letting CI run your tests/generate failures rather than running locally.
